core/src/Kokkos_Array.hpp

```diff
@@ -435,6 +435,44 @@ KOKKOS_FUNCTION constexpr T const&& get(Array<T, N> const&& a) noexcept {
 }  // namespace Kokkos
 //</editor-fold>
 
+//<editor-fold desc="Support for range-based for loop">
+namespace Kokkos {
+
+template <class T, std::size_t N>
+constexpr T const* begin(Array<T, N> const& a) {
```

Comment:
Shouldn't these all be decorated with `KOKKOS_FUNCTION`?

Also, the corresponding functions in `std::array` are marked `noexcept`.

Reply:
Fixed